packages/web/src/components/list/MultiDataList.js

```diff
@@ -364,28 +364,53 @@ class MultiDataList extends Component {
 
 	handleClick = (e) => {
 		const { value, onChange } = this.props;
-		const { value: listValue } = e.target;
-		if (value === undefined) {
-			this.setValue(listValue);
-		} else if (onChange) {
-			onChange(listValue);
+		if (isEvent(e)) {
```

Comment:
what's the purpose of this? This `if` condition adds to an already complicated logic flow. Can you please elaborate on why have we added this and briefly go over the updated flow for controlled/uncontrolled components?

Reply:
It's to detect that whether the passed argument is a synthetic event or not, if not then it'll treat the passed argument as value.
I'm exposing the `handleChange` method in `render` prop, there might be a case where user wants to change the value by `onClick` event rather than any `radio` or `checkbox` input, this condition handles it well.

Reply:
I feel e should always be part of the parameter, it allows the user to manipulate things based on an event object. Second args can be value, WDYT?

Reply:
That would certainly be a better spec. We can do it, but it's going to be a massive change, and we don't keep track of the events after the entry funnel of these callbacks. At times, these callbacks get triggered via changes like SelectedFilters clearing.